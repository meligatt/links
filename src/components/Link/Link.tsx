import * as React from "react";
import styled from "styled-components";

const LinkStyled = styled.a`
  border-width: 2px;
  border-style: solid;
  border-color: pink;
  background-color: beige;
  color: #000000;
  border-radius: 8px;
`;

type LinkProps = {
  href: string;
  text: string;
};

export const Link: React.FunctionComponent<LinkProps> = ({
  href,
  text,
}): JSX.Element => {
  return (
    <LinkStyled href={href} target="_blank">
      {text}
    </LinkStyled>
  );
};
