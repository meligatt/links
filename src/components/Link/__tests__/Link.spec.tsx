import * as React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Link } from "../Link";

describe("Link", () => {
  test("renders image with default props", () => {
    const { getByText } = render(
      <Link
        href="https://meligatt.github.io/melissa-gattoni-frontend-dev.jpeg"
        text="mocked text"
      />
    );
    expect(getByText("mocked text")).toMatchSnapshot();
  });
});
