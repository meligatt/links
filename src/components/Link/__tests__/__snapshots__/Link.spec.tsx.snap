// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders image with default props 1`] = `
<a
  class="sc-AxjAm loAriE"
  href="https://meligatt.github.io/melissa-gattoni-frontend-dev.jpeg"
  target="_blank"
>
  mocked text
</a>
`;
